lib/libutils/isoc/bget_malloc.c

```diff
@@ -230,7 +230,13 @@ static void gen_malloc_get_stats(struct malloc_ctx *ctx,
 {
 	uint32_t exceptions = malloc_lock(ctx);
 
-	memcpy(stats, &ctx->mstats, sizeof(*stats));
+	/*
+	 * Not using memcpy() here due to ASAN. This file is NEVER built with
+	 * ASAN even when ASAN is enabled globally. Therefore the variable
+	 * pointed to by ctx is not instrumented. Using an instrumented function
+	 * (memcpy()) to access it would cause a panic.
+	 */
+	*stats = ctx->mstats;
```

Comment:
Please use `asan_memcpy_unchecked()` instead since the compiler is free to replace a struct assignment with a call to `memcpy()`.

Reply:
Hmm, I think we should use the same patterns as for `memset_unchecked()` in this file.

Reply:
> Please use `asan_memcpy_unchecked()` instead since the compiler is free to replace a struct assignment with a call to `memcpy()`

Is it? Oh yes, I thought about it, believed the compiler would know about the situation, but the truth is it cannot since we don't provide any ASAN flag when building this file so it's just regular compilation and optimization.

OK, I will introduce `memcpy_unchecked()`.